dist/common/scripts/scylla_io_setup

```diff
@@ -102,6 +103,30 @@ class scylla_cpuinfo:
         else:
             return len(self._cpu_data["system"])
 
+def configure_iotune_open_fd_limit(shards_count):
+    try:
+        fd_limits = resource.getrlimit(resource.RLIMIT_NOFILE)
+    except (OSError, ValueError) as e:
+        logging.warning("Could not get the limit of count of open file descriptors!")
+        logging.warning("iotune will proceed with the default limit. This may cause problems.")
+        return
+
+    soft_limit, hard_limit = fd_limits
+    precalculated_fds_count = (8 * shards_count) + 500
+
+    # If hard_limit is greater than precalculated_fds_count set it as soft and as hard limit.
+    required_fds_count = max(hard_limit, precalculated_fds_count)
+    soft_limit = max(soft_limit, required_fds_count)
+    hard_limit = max(hard_limit, required_fds_count)
```

Comment:
`resource.RLIM_INFINITY == -1`

If hard_limit is unlimited on the target machine, then this logic will not work.

Reply:
It does not seem to be a possible case:

https://unix.stackexchange.com/questions/674218/can-you-set-rlimit-nofile-to-rlim-infinity

https://stackoverflow.com/questions/1212925/on-linux-set-maximum-open-files-to-unlimited-possible

Reply:
```c
# define RLIM_INFINITY      (~0UL)
```

It's intentionally unsigned in C. I wonder how

```python
    if value < resource.RLIM_INFINITY:
        ...
```

was ever supposed to work in Python

Reply:
It would not work - I tested it manually.

Just to be on the safe side I added a logic to handle the case when `resource.RLIM_INFINITY` is returned by `getrlimit()` as hard limit.